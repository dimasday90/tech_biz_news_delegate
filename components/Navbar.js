import { Navbar, Nav, Button } from "react-bootstrap";
import { useRouter } from "next/router";
import ReactLoading from "react-loading";

export default function Appbar() {
  const [loading, setLoading] = React.useState(false);
  const router = useRouter();
  const btnStyle = {
    marginRight: 10
  };
  const handleClick = async (event, link) => {
    await event.preventDefault();
    await setLoading(true);
    await router.push(link);
    await setLoading(false);
  };
  return (
    <Navbar
      collapseOnSelect
      expand="md"
      bg="dark"
      variant="dark"
      className="justify-content-between"
      fixed="top"
      style={{ minHeight: 77 }}
    >
      <Navbar.Brand>
        {loading ? (
          <ReactLoading type="cubes" color="white" height={50} width={50} />
        ) : (
          "Tech Biz News"
        )}
      </Navbar.Brand>
      <Navbar.Toggle aria-controls="responsive-navbar-nav" />
      <Navbar.Collapse id="responsive-navbar-nav">
        <Nav className="mr-auto">
          <Nav.Item>
            <Button
              variant={router.pathname === "/" ? "light" : "outline-light"}
              style={btnStyle}
              onClick={event => handleClick(event, "/")}
            >
              Home
            </Button>
          </Nav.Item>
          <Nav.Item>
            <Button
              variant={router.pathname === "/tech" ? "light" : "outline-light"}
              style={btnStyle}
              onClick={event => handleClick(event, "/tech")}
            >
              Technology
            </Button>
          </Nav.Item>
          <Nav.Item>
            <Button
              variant={
                router.pathname === "/business" ? "light" : "outline-light"
              }
              style={btnStyle}
              onClick={event => handleClick(event, "/business")}
            >
              Business
            </Button>
          </Nav.Item>
        </Nav>
      </Navbar.Collapse>
    </Navbar>
  );
}
