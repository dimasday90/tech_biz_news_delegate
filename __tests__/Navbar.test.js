import React from "react";
import { render, cleanup } from "@testing-library/react";
import * as nextRouter from "next/router";
import renderer from "react-test-renderer";

import Navbar from "../components/Navbar";

afterEach(cleanup);

describe("Navbar Test", () => {
  test("should render navbar text", () => {
    nextRouter.useRouter = jest.fn();
    nextRouter.useRouter.mockImplementation(() => ({ route: "/" }));
    const { getByText } = render(<Navbar />);
    expect(getByText("Tech Biz News")).toBeInTheDocument();
    expect(getByText("Home")).toBeInTheDocument();
    expect(getByText("Technology")).toBeInTheDocument();
    expect(getByText("Business")).toBeInTheDocument();
  });
});

describe("With Snapshot Testing", () => {
  it("should match navbar coding", () => {
    nextRouter.useRouter = jest.fn();
    nextRouter.useRouter.mockImplementation(() => ({ route: "/" }));
    const component = renderer.create(<Navbar />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
