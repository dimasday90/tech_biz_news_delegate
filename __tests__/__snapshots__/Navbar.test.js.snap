// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing should match navbar coding 1`] = `
<nav
  className="justify-content-between navbar navbar-expand-md navbar-dark bg-dark fixed-top"
  style={
    Object {
      "minHeight": 77,
    }
  }
>
  <span
    className="navbar-brand"
  >
    Tech Biz News
  </span>
  <button
    aria-controls="responsive-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="responsive-navbar-nav"
  >
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <div
        className="nav-item"
      >
        <button
          className="btn btn-outline-light"
          disabled={false}
          onClick={[Function]}
          style={
            Object {
              "marginRight": 10,
            }
          }
          type="button"
        >
          Home
        </button>
      </div>
      <div
        className="nav-item"
      >
        <button
          className="btn btn-outline-light"
          disabled={false}
          onClick={[Function]}
          style={
            Object {
              "marginRight": 10,
            }
          }
          type="button"
        >
          Technology
        </button>
      </div>
      <div
        className="nav-item"
      >
        <button
          className="btn btn-outline-light"
          disabled={false}
          onClick={[Function]}
          style={
            Object {
              "marginRight": 10,
            }
          }
          type="button"
        >
          Business
        </button>
      </div>
    </div>
  </div>
</nav>
`;
